script.lua

```diff
@@ -7192,13 +7192,18 @@ function swapPlayerAreaObjects(a, b, colorA, colorB)
                     table.insert(t, obj)
                 end
             end
+            for _,obj in ipairs(getObjects()) do
+                if obj.type == "Fog" and obj.getData().FogColor == color and obj.getPosition().x == tables[color].getPosition().x then
```

Comment:
I'm checking not just the colour of hiding zones here, but also their x position. This is to make it only update hiding zones that are associated with a player area. The idea being that some custom content might create loose hiding zones (over the island, say), that wouldn't want shifting whenever players swap seats.

The hiding zones for the version of Destiny Unfolds I'm working on are all associated with player tables. There are two per player, once coincident with the destiny hand, and one covering the power gain area. As such, they both have the same x coordinate as the player table, and this works properly for them.

However, this is a slightly fragile check, as some other hiding zone *could*, by coincidence, have the same x coordinate. It's unlikely, given the precision of floating point numbers (I'm safe to do a floating point comparison without an epsilon here as the x coordinates of hands and the power gain area are copied directly from that of the player table), but it's theoretically possible.

I'm up for removing this part of the check if you'd just prefer not to support hiding zones that aren't associated with player areas, or up for replacing it with a better check if you can think of a better method.

Reply:
hmm to make this a little less fragile maybe you just check to make sure the x coordinate is greater than (or less than depending upon which way x counts) or equal to the table's x (with maybe some offset)

Reply:
Sorry, I don't think I understand what you're suggesting.

Here's an image to explain what I'm trying to do:
![image](https://github.com/spirit-island/spirit-island-mod/assets/19480435/3194f29f-cc0c-4a84-bb0a-e91643500df0)

The hiding zones covering the pick power zones and the fourth hand are used by Destiny Unfolds, and should move with seat swaps. The one over the island is some hypothetical thing that might be added by another piece of custom content, and shouldn't move with seat swaps (but should still recolour). Checking for equality of the x coordinate makes this work properly (with that very slight risk of false positives), but checking if it's greater (or less) than or equal would, in this particular, make whether it moves depend upon what colour it is: here it wouldn't move if Downpour swaps seats but if it was red, it would move if Vengeance swaps seats.

Reply:
sorry guess i was getting things backwards. what i meant is that we want to make sure we aren't moving things around that above the power card area, since that stuff is generally outside of the bounds of the player area (though i guess if a player added extra seats it could be?)

Reply:
basically i want this code to not be as fragile

Reply:
Ah, right, the z variable. Yup, that works; I can add that in. If I make it such that it can't be *further from the player table* than the power card area, then that'll still work with extra added seats.